### PR TITLE
https://github.com/alibaba/nacos/issues/11250

### PR DESCRIPTION
**Pull request by llm163520**
参考Log4J2NacosLogging处理，当location不存在时则不加载

Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by llm163520** [comment]
image

**Comment by KomachiSion** [comment]
我没理解这么修改的原因。

**Comment by llm163520** [comment]
本地已经有自己日志配置了，不需要加载nacos-logback.xml配置文件呢

**Comment by llm163520** [comment]
我理解也并有没强制调整这块，但是从代码的健壮性上来说；指定这个肯定不存在，没必要去执行这块逻辑呢
该处代码有三个问题；
1、存在越界的情况，get(0)？，建议增加判断
2、地址不存在还执行加载，建议增加判断空或不存在则不执行
3、日志版本的设定与获取的判断方式存在问题；这块建议查看下逻辑，应该是使用判断方式有误

**Comment by KomachiSion** [comment]
所以你的意思是你只是做了重构，没有修改逻辑是吗？

File: client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java [hunk 1]
```diff
@@ -77,10 +78,17 @@ private LoggerContext loadConfigurationOnStart() {
...
-            nacosLogbackConfigurator.setContext(loggerContext);
-            nacosLogbackConfigurator.configure(ResourceUtils.getResourceUrl(location));
+            nacosLogbackConfigurators.stream()
+                    .filter(c -> c.getVersion() == userVersion).findFirst().ifPresent(nacosLogbackConfigurator->{
+                        nacosLogbackConfigurator.setContext(loggerContext);
+                        if(StringUtils.isNotBlank(location)){
+                            try {
+                                nacosLogbackConfigurator.configure(ResourceUtils.getResourceUrl(location));
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:88]: 不需要包装异常，要包装至少包装成NacosException

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by llm163520** [comment]
可以这么理解的

File: client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java (hunk 1 above)
Comment (llm163520) [reply client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:88]: 这块异常要往外抛的话；需要抛运行时异常(RuntimeException)，要不就内部输出wran日志

Comment (llm163520) [reply client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:88]: 已参考之前的异常调整异常抛出内容；
throw new IllegalStateException("Could not initialize Logback Nacos logging from " + location, e);

**Comment by llm163520** [comment]
当我本地不存在nacos-logback.xml配置文件时，会触发
`WARN  2023-10-23 10:27:36.933 [-main][*][c.a.n.c.l.NacosLogging]: Load Logback Configuration of Nacos fail, message: Could not initialize Logback Nacos logging from classpath:nacos-logback.xml`
image

**Comment by KomachiSion** [comment]
> 当我本地不存在nacos-logback.xml配置文件时，会触发 `WARN 2023-10-23 10:27:36.933 [-main][*][c.a.n.c.l.NacosLogging]: Load Logback Configuration of Nacos fail, message: Could not initialize Logback Nacos logging from classpath:nacos-logback.xml` image

为什么会不存在， nacos-client的jar包里封装了这个文件的

File: client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java [hunk 2]
```diff
@@ -73,18 +75,22 @@ private boolean hasListener(LoggerContext loggerContext) {
...
+                NacosLogbackConfigurator.class);
+        nacosLogbackConfigurators.stream()
+                .filter(c -> c.getVersion() == userVersion).findFirst().ifPresent(nacosLogbackConfigurator->{
+                    nacosLogbackConfigurator.setContext(loggerContext);
+                    if(StringUtils.isNotBlank(location)){
+                        try {
+                            nacosLogbackConfigurator.configure(ResourceUtils.getResourceUrl(location));
+                        } catch (Exception e) {
+                            throw new IllegalStateException("Could not initialize Logback Nacos logging from " + location, e);
+                            //throw new NacosException(NacosException.RESOURCE_NOT_FOUND, e);
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:89]: 不需要的代码无特殊情况直接删除，不需要注视掉

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
> > > 当我本地不存在nacos-logback.xml配置文件时，会触发 `WARN 2023-10-23 10:27:36.933 [-main][*][c.a.n.c.l.NacosLogging]: Load Logback Configuration of Nacos fail, message: Could not initialize Logback Nacos logging from classpath:nacos-logback.xml` image
> > 
> > 
> > 为什么会不存在， nacos-client的jar包里封装了这个文件的
> 
> 估摸着读取的方式不对，通过日志可以看出实际并未加载到

有空可以研究下，为什么没读取到，感觉不太应该。

File: client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java (hunk 2 above)
Comment (llm163520) [reply client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:89]: 好的，已处理并提交

File: client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java [hunk 3]
```diff
@@ -25,24 +25,24 @@
...
  * Support for Logback version 1.0.8 or higher
  *
  * @author <a href="mailto:huangxiaoyu1018@gmail.com">hxy1991</a>
  * @author <a href="mailto:hujun3@xiaomi.com">hujun</a>
- *
  * @since 0.9.0
  */
 public class LogbackNacosLogging extends AbstractNacosLogging {
-    
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/logging/logback/LogbackNacosLogging.java:41]: Please don't change the indent, please use nacos code style to reformat code.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by KomachiSion** [comment]
@llm163520 , Your PR change the indents which is not need change and no relative with the issue, Please fix it. 

I see other problem all has been fixed. Only the indents problem need fix.

**Comment by llm163520** [comment]
> @llm163520 , Your PR change the indents which is not need change and no relative with the issue, Please fix it.
> 
> I see other problem all has been fixed. Only the indents problem need fix.

咱们代码格式化模板是用的？

**Comment by llm163520** [comment]
> > @llm163520 , Your PR change the indents which is not need change and no relative with the issue, Please fix it.
> > I see other problem all has been fixed. Only the indents problem need fix.
> 
> 咱们代码格式化模板是用的？

忽略，已按模板格式化并重新提交

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**